release/README.md

```diff
@@ -12,11 +12,49 @@ Istio release is currently composed of artifacts for the following repos:
 
 The release consists in retagging the artifacts and creating new annotated tags.
```

Comment:
This document deserves a table of contents. See istio/istio/README.md for an example TOC format.